src/Avalonia.Base/Media/RenderOptions.cs

```diff
@@ -1,36 +1,131 @@
 ﻿using Avalonia.Media.Imaging;
 
 namespace Avalonia.Media
-{ 
-    public class RenderOptions
+{
+    public readonly record struct RenderOptions
     {
+        public BitmapInterpolationMode BitmapInterpolationMode { get; init; }
+        public EdgeMode EdgeMode { get; init; }
+        public TextRenderingMode TextRenderingMode { get; init; }
+        public BitmapBlendingMode BitmapBlendingMode { get; init; }
+
+        /// <summary>
+        /// Gets the value of the BitmapInterpolationMode attached property for a visual.
+        /// </summary>
+        /// <param name="visual">The control.</param>
+        /// <returns>The control's left coordinate.</returns>
+        public static BitmapInterpolationMode GetBitmapInterpolationMode(Visual visual)
+        {
+            return visual.RenderOptions.BitmapInterpolationMode;
+        }
+
         /// <summary>
-        /// Defines the <see cref="BitmapInterpolationMode"/> property.
+        /// Sets the value of the BitmapInterpolationMode attached property for a visual.
         /// </summary>
-        public static readonly StyledProperty<BitmapInterpolationMode> BitmapInterpolationModeProperty =
-            AvaloniaProperty.RegisterAttached<RenderOptions, AvaloniaObject, BitmapInterpolationMode>(
-                "BitmapInterpolationMode", 
-                BitmapInterpolationMode.MediumQuality,
-                inherits: true);
+        /// <param name="visual">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetBitmapInterpolationMode(Visual visual, BitmapInterpolationMode value)
+        {
+            visual.RenderOptions = visual.RenderOptions with { BitmapInterpolationMode = value };
+        }
```

Comment:
Before it was bindable, but not anymore. Just want to confirm it's an expected change.

Reply:
This is expected for performance reasons. These static helpers are just for XamlX so one can set a specific component of the RenderOptions

Reply:
In WPF `RenderOptions.BitmapScalingMode` is a proper bindable property.